code/game/machinery/dna_infuser/dna_infuser.dm

```diff
@@ -52,14 +52,20 @@
 		balloon_alert(user, "not while it's on!")
 		return
 	if(occupant && infusing_from)
-		balloon_alert(user, "starting DNA infusion...")
-		start_infuse()
+		// Abort infusion if the occupant lacks valid DNA.
+		if(is_valid_occupant(occupant))
+			balloon_alert(user, "starting DNA infusion...")
+			start_infuse()
+		else
+			balloon_alert(user, "error!")
+			visible_message(span_notice("[src] displays a message on its status screen: ") + span_warning("Patient DNA is missing, corrupted, or incompatible."))
```

Comment:
```suggestion
			balloon_alert(user, "dna is missing or incompatible!")
```

ideally, it would say what the specific problem is, and we only show that

Reply:
oh you made these procs, so yes definitely split them up into two and make a unique balloon alert for each

Reply:
Fixed by commit 2d56a218ed5564976fe63be186d738470c6ef734